src/cargo/util/hex.rs

```diff
@@ -21,6 +23,19 @@ pub fn hash_u64<H: Hash>(hashable: H) -> u64 {
     hasher.finish()
 }
 
+pub fn hash_u64_file(mut file: &File) -> u64 {
+    let mut hasher = SipHasher::new_with_keys(0, 0);
+    let mut buf = [0; 64 * 1024];
+    loop {
+        let n = file.read(&mut buf).unwrap();
```

Comment:
The original code returns errors, instead of panicing.

Reply:
Yeah, I have changed it but now all the tests are failing..